src/index.js

```diff
@@ -1,10 +1,34 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { createStore } from 'redux';
+import { Provider } from 'react-redux';
+import rootReducer from './reducers'
+
 import './index.css';
-import App from './App';
+import App from './containers/App';
 import * as serviceWorker from './serviceWorker';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+
+const generateID = () => {
```

Comment:
This function `generateID` is also used in `BooksForm`. Therefore, to avoid duplication we should put it in a separate file in a `utils` folder.